collections/src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -220,6 +221,112 @@ sealed abstract class LazyList[+A] extends LinearSeq[A] with LazyListOps[A, Lazy
     if (this.isEmpty) z
     else tail.foldLeft(op(z, head))(op)
   }
+
+  /** Write all defined elements of this iterable into given string builder.
+    *  The written text begins with the string `start` and is finished by the string
+    *  `end`. Inside, the string representations of defined elements (w.r.t.
+    *  the method `toString()`) are separated by the string `sep`. The method will
+    *  not force evaluation of undefined elements. A tail of such elements will be
+    * represented by a `"?"` instead.  A cyclic stream is represented by a `"..."`
+    * at the point where the cycle repeats.
+    *
+    * @param b The [[collection.mutable.StringBuilder]] factory to which we need
+    * to add the string elements.
+    * @param start The prefix of the resulting string (e.g. "LazyList(")
+    * @param sep The separator between elements of the resulting string (e.g. ",")
+    * @param end The end of the resulting string (e.g. ")")
+    * @return The original [[collection.mutable.StringBuilder]] containing the
+    * resulting string.
+    */
+  protected def toStringBuilder(b: StringBuilder, start: String, sep: String, end: String): StringBuilder = {
+    b append start
+    if (!isEmpty) {
+      if (headDefined && !tailDefined) b append head // For LazyList, head is also Lazy. Hence adds `?` if it not defined yet
+      if (!headDefined && tailDefined) b append "?" // For LazyList, head is also Lazy. Hence adds `?` if it not defined yet
+      if (headDefined && tailDefined) b append head
+      if (!headDefined && !tailDefined) {
+        // nothing to evaluate. Bail out early
+        b append "?" append end
+        return b
+      }
+      var cursor = this
+      var n = 1
+      if (cursor.tailDefined) {  // If tailDefined, also !isEmpty
+        var scout = cursor.tail
+        if (scout.isEmpty) {
+          // Single element.  Bail out early.
+          return b
```

Comment:
`If (headDefined && tailDefined && tail.isEmpty` then result is `LazyList("head"` - terminating parenthesis is absent.

Reply:
@optician , I fixed this. Yeah you are right, it was not adding `end` parenthesis there.

Reply:
Have added corresponding unit test for it as well